shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -230,12 +233,27 @@ protected void encode(ChannelHandlerContext ctx, Object msg, ByteBuf out) throws
             SetupAppend setup = (SetupAppend) msg;
             setupSegments.put(new SimpleImmutableEntry<>(setup.getSegment(), setup.getWriterId()),
                               new Session(setup.getWriterId(), setup.getRequestId()));
+        } else if (msg instanceof CloseAppend) {
```

Comment:
There are some commonalities between CloseAppend, BlockTimeout, and Flush. In all cases all appends for a particular writerId. Let's make a private method for that.

Reply:
Not exactly  100%; However,  I have reordered them now..
1) close append
       - break the existing append
        - flush the pending session data
       - remove the entry from setupSegments.
       - send the wire command
2)  flush command
       - if the session has any data ( if the give session of writer id and segment does not have any data, then no need to break the append)  then break the append
       - flush the pending data
                 NOTE : no wire command to segment store
3) block timeout
        -  break the append
        -  write all pending data
             NOTE : no wire command to segment store.

you have mentioned " In all cases all appends for a particular writerId" 
   for this we need type cast to  some message type ..
please let me know your thoughts